Week2/homework/part1/queries.js

```diff
@@ -0,0 +1,123 @@
+      '\n' +
+      '\n' +
+      '*****************************************************************************************';
+
+    console.log(message);
+    connection.end();
+  },
+  learnCapital: countryName => {
+    connection.connect();
+    connection.execute(
+      'SELECT city.Name FROM city INNER JOIN country ON country.Capital = city.ID WHERE country.Name = ?',
+      [countryName],
+      (err, results, fields) => {
+        if (err) throw err;
+        console.log(`The capital of ${countryName} is ${results[0].Name}`);
+      }
+    );
+    connection.end();
+  },
+  learnSpokenLanguage: regionName => {
+    connection.connect();
+    connection.execute(
+      'SELECT DISTINCT countrylanguage.Language FROM countrylanguage JOIN country ON country.Code = countrylanguage.CountryCode AND country.Region = ?',
+      [regionName],
+      (err, results, fields) => {
+        if (err) throw err;
+        console.log(
+          `The spoken languages in ${regionName} region are: ${Array.from(results).map(
+            language => '\n' + language.Language
+          )}`
+        );
+      }
+    );
+    connection.end();
+  },
+  findCities: languageName => {
+    connection.connect();
+    connection.execute(
+      'SELECT DISTINCT COUNT(city.ID) FROM city JOIN countrylanguage ON city.CountryCode = countrylanguage.CountryCode AND countrylanguage.Language = ?',
+      [languageName],
+      (err, results, fields) => {
+        if (err) throw err;
+        console.log(
+          `The number of cities in which ${languageName} is spoken: ${Object.values(results[0])}`
+        );
+      }
+    );
+    connection.end();
+  },
+  findSimilarCountriesByOfficialLanguage: () => {
+    connection.connect();
+    connection.query(
+      'SELECT DISTINCT country.Name FROM countrylanguage A, countrylanguage B, country WHERE A.CountryCode = country.Code AND A.IsOfficial = "T" AND A.Language = B.Language',
+      (err, results) => {
+        if (err) throw err;
+        console.log(results);
+        const countryNames = Array.from(results).map(country => '\n' + country.Name);
+        console.log(
+          `All of these countries have official language that at least a different country has the same official language: ${countryNames}`
+        );
+      }
+    );
+    connection.end();
+  },
+  findSimilarCountriesByRegion: () => {
+    connection.connect();
+    connection.query(
+      'SELECT DISTINCT A.Name, A.Region FROM country A, country B WHERE A.Region = B.Region',
+      (err, results, fields) => {
+        if (err) throw err;
+        console.log(results);
+      }
+    );
+    connection.end();
+  },
+  listContinents: () => {
+    connection.connect();
+    connection.query(
+      'SELECT DISTINCT Continent, COUNT(Language) AS Language FROM country INNER JOIN countrylanguage ON country.Code = countrylanguage.CountryCode GROUP BY Continent',
```

Comment:
Are you sure this is the correct query?
This is returning the following result:
North America : 109,
Asia : 239,
Africa : 310,
Europe : 202,
South America : 42,
Oceania : 82